src/main/PresentationExchange.ts

```diff
@@ -151,16 +151,16 @@ export class PresentationExchange {
       }
     }
 
-    function addSingleIdTokenPDToPDs(definition) {
-      PresentationExchange.assertValidPresentationDefinition(definition.value);
-      if (!definition.path.includes(PresentationLocation.ID_TOKEN)) {
-        throw new Error(SIOPErrors.REQUEST_CLAIMS_PRESENTATION_DEFINITION_NOT_VALID);
-      }
-      allDefinitions.push({ definition: definition.value, location: PresentationLocation.ID_TOKEN });
+    function addSingleIdTokenPDToPDs(definition: IPresentationDefinition): void {
+      PresentationExchange.assertValidPresentationDefinition(definition);
+      allDefinitions.push({ definition: definition, location: PresentationLocation.ID_TOKEN });
     }
 
     async function extractPDFromOtherTokens() {
-      const definitions = extractDataFromPath(obj, '$..verifiable_presentations.presentation_definition');
+      const definitions: PresentationDefinitionV1[] | PresentationDefinitionV2[] = extractDataFromPath(
```

Comment:
same as above, can you add a test case for this?

Reply:
Without an explicit check it will not help, at runtime it can be anything.